python/kubeflow/trainer/api/trainer_client.py

```diff
@@ -489,36 +557,28 @@ def wait_for_job_status(
                 f"Expected status {status} must be a subset of {job_statuses}"
             )
 
-        # Use Kubernetes watch API to monitor the TrainJob's Pods.
-        w = watch.Watch()
-        try:
-            for event in w.stream(
-                self.core_api.list_namespaced_pod,
-                self.namespace,
-                label_selector=constants.POD_LABEL_SELECTOR.format(trainjob_name=name),
-                timeout_seconds=timeout,
-            ):
-                # Check the status after event is generated for the TrainJob's Pods.
-                trainjob = self.get_job(name)
-                logger.debug(f"TrainJob {name}, status {trainjob.status}")
+        if polling_interval > timeout:
```

Comment:
@astefanutti I have to refactor the `wait_for_job_status()` API to perform polling as before.
The problem that I saw is when Pods are succeeded too fast, and TrainJob controller doesn't [add the Complete condition to the `.status.conditions`.](https://github.com/kubeflow/sdk/blob/b435e07d860348965a9e79023e4e9705e6cd5992/python/kubeflow/trainer/api/trainer_client.py#L747-L752) 

Since we only watch for Pod events, we can't catch this event, and TrainJob is stuck in Running condition.

Alternatively, we can watch both for TrainJob + Pods with two Python threads, but I am not sure if it worths it.

What do you think @astefanutti ?